Controllers/CustomerController.cs

```diff
@@ -147,7 +156,89 @@ public string Delete(IFormCollection collection)
+        public IActionResult GetCsvData(){
+            try{
+                var path = "C:\\Users\\Nikhil\\Downloads\\Northwind_database_csv\\customer_custom.csv";
+
+                using (StreamReader reader = new StreamReader(path))
+                using (CsvReader csv_reader = new CsvReader(reader, CultureInfo.InvariantCulture))
+                {
+                    var records = csv_reader.GetRecords<Customer>().ToList();
+
+                    return Ok(records);
+                }
+            }
+            catch(Exception e){
+                throw;
+            }
+        }
+
+        // POST: CustomerController/Csv/post
+        [HttpPost("Csv/post")]
+        public IActionResult WriteCsvData(Customer customer){
+            try{
+                var path = "C:\\Users\\Nikhil\\Downloads\\Northwind_database_csv\\customer_custom.csv";
+
+                // Append to the file.
+                var config = new CsvConfiguration(CultureInfo.InvariantCulture)
+                {
+                    // Don't write the header again.
+                    HasHeaderRecord = false,
+                };
+                using(Stream stream = System.IO.File.Open(path, FileMode.Append))
+                using(StreamWriter sw = new StreamWriter(stream))
+                using(CsvWriter csv_writer = new CsvWriter(sw, config)){
+                    csv_writer.WriteRecord<Customer>(customer);
+
+                    return Ok(customer);
+                }
+            }
+            catch(Exception e){
+                throw;
+            }
+        }
+
+        // UPDATE: Conterollers/Csv/UpdateToDatabase
+        [HttpPost("Csv/CsvToDatabase")]
+        public IActionResult CsvToDatabase(string path){
+            try
+            {
+                using (StreamReader reader = new StreamReader(path))
+                using (CsvReader csv_reader = new CsvReader(reader, CultureInfo.InvariantCulture))
+                {
+                    var record = new Customer();
+                    var records = csv_reader.GetRecords<Customer>().ToList();
+
+                    foreach(var r in records)
+                    {
+                        Customer obj = new Customer();
+                        obj.Address = r.Address;
+                        obj.City = r.City;
+                        obj.CompanyName = r.CompanyName;
+                        obj.Country = r.Country;
+                        obj.Fax = r.Fax;
+                        obj.Phone = r.Phone;
+                        obj.PostalCode = r.PostalCode;
+                        obj.Region = r.Region;
+
+                        _db.Customer.Add(obj);
+                    }
+                    _db.SaveChanges();
+
+                    records = csv_reader.GetRecords<Customer>().ToList();
+                    return Ok(records);
+                }
+            }
+            catch (Exception e)
+            {
+                throw;
+                return NoContent();
```

Comment:
I think throw is going to stop the program, so return statement won't be able to execute.

Reply:
Yes